File: solve/0910_문자열 나누기/postforty/문자열 나누기.js
// https://school.programmers.co.kr/learn/courses/30/lessons/140108?language=javascript
function solution(s) {
  var answer = 0;
  let x = s[0];
  let countX = 0;
  let countNotX = 0;

  for (let c in s) {
    if (countX !== 0 && countX === countNotX) {
      answer++;
      countX = 0;
      countNotX = 0;
      x = s[c];
    }

    countX += x === s[c] && 1;
    countNotX += x !== s[c] && 1;
  }

  answer += (countX !== 0 || countNotX !== 0) && 1;
  return answer;
}

console.log(solution("abracadabra")); // 6

File: solve/0910_문자열 나누기/postforty/문자열 나누기.py
# https://school.programmers.co.kr/learn/courses/30/lessons/140108?language=python3
def solution(s):
    answer = 0
    x = s[0]
    count_x = 0
    count_not_x = 0

    for c in s:
        if count_x != 0 and count_x == count_not_x:
            answer += 1
            count_x = 0
            count_not_x = 0
            x = c

        count_x += 1 if x == c else 0
        count_not_x += 1 if x != c else 0

    answer += 1 if count_x != 0 or count_not_x != 0 else 0

    return answer


print(solution("abracadabra"))  # 6
